examples/pytorch/argo/main.py

```diff
@@ -0,0 +1,238 @@
+        else:
+            raise NotImplementedError
+
+        self.dropout = nn.Dropout(0.5)
+        self.hid_size = hid_size
+        self.out_size = out_size
+
+    def forward(self, blocks, x):
+        h = x
+        if hasattr(blocks, '__len__'):
+            for l, (layer, block) in enumerate(zip(self.layers, blocks)):
+                h = layer(block, h)
+                if l != len(self.layers) - 1:
+                    h = F.relu(h)
+                    h = self.dropout(h)
+        else:
+            for l, layer in enumerate(self.layers):
+                h = layer(blocks, h)
+                if l != len(self.layers) - 1:
+                    h = F.relu(h)
+                    h = self.dropout(h)
+        return h
+
+
+def _train(**kwargs):
+    total_loss = 0
+    loader = kwargs['loader']
+    model = kwargs['model']
+    opt = kwargs['opt']
+    load_core = kwargs['load_core']
+    comp_core = kwargs['comp_core']
+
+    device = torch.device("cpu")
+    with loader.enable_cpu_affinity(loader_cores=load_core, compute_cores=comp_core):
+        for it, (input_nodes, output_nodes, blocks) in enumerate(loader):
+            if hasattr(blocks, '__len__'):
+                x = blocks[0].srcdata["feat"].to(torch.float32)
+                y = blocks[-1].dstdata["label"]
+            else:
+                x = blocks.srcdata["feat"].to(torch.float32)
+                y = blocks.dstdata["label"]
+            if kwargs['device'] == "cpu":  # for papers100M
+                y = y.type(torch.LongTensor)
+                y_hat = model(blocks, x)
+            else:
+                y = y.type(torch.LongTensor).to(device)
+                y_hat = model(blocks, x).to(device)
+            try:
+                loss = F.cross_entropy(y_hat[:output_nodes.shape[0]], y[:output_nodes.shape[0]])
+            except:
+                loss = F.binary_cross_entropy_with_logits(
+                    y_hat[:output_nodes.shape[0]].float(),
+                    y[:output_nodes.shape[0]].float(),
+                    reduction="sum"
+                )  
+            opt.zero_grad()
+            loss.backward()
+            opt.step()
+            del input_nodes, output_nodes, blocks
+            total_loss += loss.item()  
+    return total_loss
+
+
+def train(args, g, data, rank, world_size, comp_core, load_core, counter, b_size, ep ):
+
+    num_classes, train_idx = data
+    dist.init_process_group('gloo', rank=rank, world_size=world_size)
+    device = torch.device("cpu")
+    hidden = args.hidden
+    # create GraphSAGE model
+    in_size = g.ndata["feat"].shape[1]
+    model = GNN(in_size, hidden, num_classes,
+                num_layers=args.layer, model_name=args.model).to(device)
+    model = DistributedDataParallel(model)
+    num_of_samplers = len(load_core)
+    # create loader
+    drop_last, shuffle = True, True
+    if args.sampler.lower() == 'neighbor':
+        sampler = NeighborSampler(
+            [15, 10, 5],
```

Comment:
Maybe we could delete this [note](https://github.com/dmlc/dgl/blob/a16d182c6c756e40adb0de7c84f97b4b6b4f426d/examples/pytorch/argo/README.md?plain=1#L49) from readme and make it to also be a parameter in a similar manner it is done here: https://github.com/dmlc/dgl/blob/a16d182c6c756e40adb0de7c84f97b4b6b4f426d/examples/pytorch/argo/ogb_example.py#L245

Reply:
I agree. I've added the fanout argument and deleted the comment in README.